File: problemas/1100/1160.dart
import 'dart:io';

void main() {
  final t = int.parse(stdin.readLineSync());
  String tempo;

  for (var i = 0; i < t; i++) {
    final entrada = stdin.readLineSync().split(' ');
    double pa = double.parse(entrada[0]);
    double pb = double.parse(entrada[1]);
    final g1 = double.parse(entrada[2]) / 100;
    final g2 = double.parse(entrada[3]) / 100;

    tempo = 'Mais de 1 seculo.';

    for (var j = 1; j < 101; j++) {
      pa += (pa * g1).floor();
      pb += (pb * g2).floor();

      if (pa > pb) {
        tempo = '$j anos.';
        break;
      }
    }

    stdout.writeln(tempo);
  }
}

File: problemas/1100/1160.php
<?php

$t = readline();

for ($i = 0; $i < $t; $i++) {
    [$pa, $pb, $g1, $g2] = explode(' ', readline());
    $g1 /= 100;
    $g2 /= 100;
    $tempo = 'Mais de 1 seculo.';

    for ($j = 1; $j < 101; $j++) {
        $pa += floor($pa * $g1);
        $pb += floor($pb * $g2);

        if ($pa > $pb) {
            $tempo = "{$j} anos.";
            break;
        }
    }

    echo $tempo . "\n";
}